src/Language/PureScript/TypeChecker.hs

```diff
@@ -163,10 +163,12 @@ addTypeClass _ qualifiedClassName args implies dependencies ds kind = do
     classMembers :: [(Ident, SourceType)]
     classMembers = map toPair ds
 
-    mkNewClass :: Environment -> TypeClassData
-    mkNewClass env = makeTypeClassData args classMembers implies dependencies ctIsEmpty
+    mkNewClass :: Environment -> m TypeClassData
+    mkNewClass env = do
+      implies' <- (traverse . overConstraintArgs . traverse) replaceAllTypeSynonyms implies
```

Comment:
This is pretty minor, but it's a little odd to be both passing in an explicit `env` as a parameter and pulling an implicit `Environment` out of the monad in `replaceAllTypeSynonyms`—in current usage they'll always be the same `Environment`, but that fact isn't immediately obvious. Since you're adding the monad to `mkNewClass`, maybe get rid of the `env` parameter, use `getEnv` inside, and pass it to `findSuperClass` or something like that?

Otherwise, LGTM!

Reply:
You’re right, I applied your suggestion!